catalog/core/catalog-core-camelcomponent/src/test/java/ddf/camel/component/catalog/content/ContentProducerDataAccessObjectTest.java

```diff
@@ -61,6 +63,8 @@
     ContentProducerDataAccessObject contentProducerDataAccessObject =
             new ContentProducerDataAccessObject();
 
+    private FilterBuilder filterBuilder = new GeotoolsFilterBuilder();
```

Comment:
We should add a `testUpdateContentItem` that asserts that the existing metacard is looked up and used for the update, and that the attribute overrides from the create are not overwritten. We should also cover any negative cases for the conditional branches.

Likewise, we should update the existing integration test to verify that attribute overrides are preserved on update: https://github.com/codice/ddf/blob/ce0646e54dede07c57bf939a70c27e68a313a844/distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestCatalog.java#L1514-L1514